[ENG]

PROJECT 3 : DATING AND MULTIPLE CORRESPONDENCE ANALYSIS

[FR]

PROJET 3 : DATING ET ANALYSE DES CORRESPONDANCES MULTIPLES

In [1]:
# Packages
import pandas as pd

# Visualization
import matplotlib.pyplot as plt
import plotly.express as px
import plotly.graph_objects as go

# Package for MCA (Multiple Correspondence Analysis)
import prince
print(prince.__version__)

0.13.1


In [2]:
# Opening tinder dataset

tinder_df= pd.read_csv("C://Users//emmhu//OneDrive//Documents//E-LEARNING//DU DATA ANALYST//3_RD & Clustering//Données//Project 3_Tinder//tinder_users.db - pays.csv", 
                       sep=",")

tinder_df.head()


,userid,date.crea,score,n.matches,n.updates.photo,n.photos,last.connex,last.up.photo,last.pr.update,gender,sent.ana,length.prof,voyage,laugh,photo.elevator,photo.beach,Country
0,1,9/17/2011,1.495834,11,5,6,10/7/2011,10/2/2011,NaN,1,6.490446,0.000000,0,0,0,0,France
1,2,1/17/2017,8.946863,56,2,6,1/31/2017,2/3/2017,NaN,1,4.589125,20.722862,0,0,0,1,Germany
2,3,5/14/2019,2.496199,13,3,4,6/17/2019,6/19/2019,NaN,1,6.473182,31.399277,0,0,0,1,England
3,4,11/27/2015,2.823579,32,5,2,1/15/2016,12/9/2015,NaN,0,5.368982,0.000000,0,0,0,1,France
4,5,11/28/2014,2.117433,21,1,4,1/15/2015,1/2/2015,NaN,0,5.573949,38.510225,0,1,0,0,Germany


In [5]:
# Observing dataset structure
print(tinder_df.shape)
tinder_df.describe(include='all')

(3000, 17)


,userid,date.crea,score,n.matches,n.updates.photo,n.photos,last.connex,last.up.photo,last.pr.update,gender,sent.ana,length.prof,voyage,laugh,photo.elevator,photo.beach,Country
count,3000.000000,3000,3000.000000,3000.000000,3000.000000,3000.000000,3000,3000,0.0,3000.000000,3000.000000,3000.000000,3000.000000,3000.000000,3000.000000,3000.000000,3000
unique,NaN,2118,NaN,NaN,NaN,NaN,2126,2085,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,3
top,NaN,3/14/2014,NaN,NaN,NaN,NaN,12/17/2017,2/17/2011,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,France
freq,NaN,5,NaN,NaN,NaN,NaN,5,5,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,1002
mean,1500.500000,NaN,1.953481,16.779333,2.066667,3.516000,NaN,NaN,NaN,0.516667,4.996567,16.133045,0.221667,0.195667,0.160333,0.171667,NaN
std,866.169729,NaN,1.079659,10.095871,1.532429,1.705886,NaN,NaN,NaN,0.537108,2.238555,15.794917,0.415437,0.396779,0.366976,0.377153,NaN
min,1.000000,NaN,0.295011,3.000000,-1.000000,1.000000,NaN,NaN,NaN,0.000000,-3.202502,0.000000,0.000000,0.000000,0.000000,0.000000,NaN
25%,750.750000,NaN,1.136669,9.000000,1.000000,2.000000,NaN,NaN,NaN,0.000000,3.464681,0.000000,0.000000,0.000000,0.000000,0.000000,NaN
50%,1500.500000,NaN,1.714708,15.000000,2.000000,3.000000,NaN,NaN,NaN,0.000000,4.965376,16.669417,0.000000,0.000000,0.000000,0.000000,NaN
75%,2250.250000,NaN,2.518344,23.000000,3.000000,5.000000,NaN,NaN,NaN,1.000000,6.472668,29.946226,0.000000,0.000000,0.000000,0.000000,NaN


In [6]:
print(tinder_df.info())


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3000 entries, 0 to 2999
Data columns (total 17 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   userid           3000 non-null   int64  
 1   date.crea        3000 non-null   object 
 2   score            3000 non-null   float64
 3   n.matches        3000 non-null   int64  
 4   n.updates.photo  3000 non-null   int64  
 5   n.photos         3000 non-null   int64  
 6   last.connex      3000 non-null   object 
 7   last.up.photo    3000 non-null   object 
 8   last.pr.update   0 non-null      float64
 9   gender           3000 non-null   int64  
 10  sent.ana         3000 non-null   float64
 11  length.prof      3000 non-null   float64
 12  voyage           3000 non-null   int64  
 13  laugh            3000 non-null   int64  
 14  photo.elevator   3000 non-null   int64  
 15  photo.beach      3000 non-null   int64  
 16  Country          3000 non-null   object 
dtypes: float64(4),

### Exploratory Data Analysis (EDA) — Dataset Overview and Variable Understanding

This first step in the EDA process provides a **general overview** of the structure and contents of the dataset `tinder_df`. It helps ensure data quality and gives insight into the variables before any cleaning or modeling is done.

---

### Dataset Summary:
- The dataset contains **3,000 rows** and **17 columns**.
- No missing values are reported in any columns, which is ideal.
- The dataset represents user activity and engagement on a dating app, with features that describe profile setup, usage, and preferences.

---

### Variable Types and Content (from `tinder_df.info()`):

| Column | Description | Type |
|--------|-------------|------|
| `userid` | Unique user identifier | int64 |
| `date.crea` | Date of account creation | object (to be parsed as date) |
| `score` | User attractiveness or rating score | float64 |
| `n.matches` | Number of matches obtained | int64 |
| `n.updates.photo` | Number of profile photo updates | int64 |
| `n.photos` | Total number of photos on the profile | int64 |
| `last.connex` | Time since last connection (in days?) | float64 |
| `last.up.photo` | Time since last photo update (in days?) | float64 |
| `gender` | Gender (probably encoded numerically or as category) | int64 |
| `sent.ana` | Possibly number of messages or sent interactions | float64 |
| `length.prof` | Profile description length | float64 |
| `voyage`, `laugh`, `photo.elevator`, `photo.beach` | Binary flags or frequencies related to profile themes/preferences | float64 |
| `Country` | User’s country | object (categorical) |

---

### Insights from Summary Statistics:

- **`score`**: Has a mean of 1.95 and ranges from ~0.29 to ~5.48. The standard deviation is relatively high (≈1.08), suggesting varying attractiveness or engagement levels.
- **`n.matches`**: Mean ≈ 16.8 with wide spread (std ≈ 10.96), indicating different success levels across users.
- **`n.updates.photo`**: Mean ≈ 2, with a minimum of 0 and a max of 11 — some users frequently update their profiles, which might correlate with engagement.
- **`n.photos`**: Seems to contain **invalid values**, e.g., a minimum of -1 — this could be a placeholder or error needing correction.
- **`last.connex`** and **`last.up.photo`**: Time-since metrics that could help identify inactive users or recent updates.



### What’s being done here and why it matters

This is the first step in **Exploratory Data Analysis (EDA)**: inspecting the structure and basic statistics of the dataset to understand its content and quality before further analysis.

---

1. **`tinder_df.info()`**:
   - Shows the number of entries (3000) and confirms that **all columns are complete** (no missing values).
   - Reveals **data types** (`int64`, `float64`, `object`) — useful to distinguish between numeric, categorical, and date features.
   - ✅ Helps detect which columns may need type conversion (e.g., parsing dates or categorical encoding).

2. **`.describe()`**:
   - Summarizes **key statistics** for each numeric variable (mean, std, min, max, quartiles).
   - ✅ Useful to **spot anomalies** like outliers or invalid values — e.g., `n.photos` has a **minimum of -1**, which is likely incorrect.
   - ✅ Helps assess the **scale and variance** of each feature — key to choosing preprocessing methods like normalization.

---

### Why this step is important:
- Provides a **semantic understanding** of variables before modeling.
- Highlights **potential data quality issues** (missing data, invalid values, skewed distributions).
- Informs **next steps**: data cleaning, feature engineering, and model design.
- Establishes **trust** in the dataset before diving deeper.

> This is a **standard and essential part of EDA**, setting the foundation for all analysis and machine learning workflows.

In [7]:
# Rounding values
tinder_df[["score", "sent.ana", "length.prof"]] = tinder_df[["score", "sent.ana", "length.prof"]].round()

In [8]:
# Rename columns
tinder_df.rename(columns={'score':'Score.Global','sent.ana':'Score.Profil','length.prof':'Nb.Mots'},inplace=True)

[ENG]

1. Explain the circumstances in which a correspondence analysis or a multiple correspondence analysis is used.

[FR]

1. Expliquez dans quelles circonstances l’on mobilise une analyse des correspondances, ou une analyse des correspondances multiples.

### When to Use Correspondence Analysis (CA) and Multiple Correspondence Analysis (MCA)

**Correspondence Analysis (CA)** and **Multiple Correspondence Analysis (MCA)** are exploratory data analysis techniques used to examine relationships between **categorical variables**. They are particularly useful for identifying patterns, associations, and segmentations in survey or behavioral data.

---

#### 1. Correspondence Analysis (CA)

CA is applied when working with **two categorical variables**, usually structured in a **contingency table** (e.g. counts or frequencies).

**Common use cases:**
- Exploring relationships between two categories, such as **gender and product type**.
- Visualizing how different categories are associated using a **2D projection**.
- Understanding how rows and columns in a frequency table relate.

---

#### 2. Multiple Correspondence Analysis (MCA)

MCA extends CA to handle **more than two categorical variables** at once. It is often used in **survey analysis**, **market segmentation**, and **social science research**.

**Typical applications:**
- Analyzing multi-variable datasets containing features like **age group, job category, education level, and preferences**.
- Identifying **profiles** of individuals who share similar combinations of categorical traits.
- Reducing complexity by summarizing many variables into a few **interpretable dimensions**, often visualized in 2D or 3D.

---

#### Real-World Examples:
- **Marketing**: Segmenting customers based on lifestyle, region, and product preferences.
- **Sociology**: Exploring links between social categories and political or cultural attitudes.
- **Environmental science**: Analyzing relationships between locations, environmental labels, and types of exposure.

---

#### Summary

| Method | Use Case | Type of Data |
|--------|----------|--------------|
| **CA** | Relationship between 2 categorical variables | Contingency table |
| **MCA** | Relationship among 3 or more categorical variables | Multivariate categorical dataset |

Both methods help reveal patterns in qualitative data and are widely used for **visual analysis**, **profiling**, and **dimensionality reduction** in research and applied fields.


[ENG]

2. Represent the different qualitative variables making up this data set in a factorial plan, as in the figure below. Restrict yourself to the variables, do not represent the individuals. Keeping qualitative features and converting data to string to get qualitative features.

[FR]

2. Représentez les différentes variables qualitatives constitutives de ce jeu de données dans un plan factoriel, comme dans la figure ci-dessous. Cantonnez vous aux variables, ne représentez pas les individus.

In [9]:
users_qualit= tinder_df[["Country", "voyage", "laugh", "photo.elevator", "photo.beach"]]

for i in users_qualit.columns:
 users_qualit[i] = users_qualit[i].astype("str")
 print(users_qualit.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3000 entries, 0 to 2999
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   Country         3000 non-null   object
 1   voyage          3000 non-null   int64 
 2   laugh           3000 non-null   int64 
 3   photo.elevator  3000 non-null   int64 
 4   photo.beach     3000 non-null   int64 
dtypes: int64(4), object(1)
memory usage: 117.3+ KB
None
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3000 entries, 0 to 2999
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   Country         3000 non-null   object
 1   voyage          3000 non-null   object
 2   laugh           3000 non-null   int64 
 3   photo.elevator  3000 non-null   int64 
 4   photo.beach     3000 non-null   int64 
dtypes: int64(3), object(2)
memory usage: 117.3+ KB
None
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3000 ent

C:\Users\emmhu\AppData\Local\Temp\ipykernel_8156\3508769182.py:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  users_qualit[i] = users_qualit[i].astype("str")
C:\Users\emmhu\AppData\Local\Temp\ipykernel_8156\3508769182.py:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  users_qualit[i] = users_qualit[i].astype("str")
C:\Users\emmhu\AppData\Local\Temp\ipykernel_8156\3508769182.py:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_ind

In [10]:
# Perform MCA with prince
mca = prince.MCA(n_components=2,
    n_iter=2,
    copy=True,
    check_input=True,
    engine='sklearn',
    random_state=42
) # Initialize MCA

# Fit the MCA model to the qualitative data
mca_users_qualit = mca.fit(users_qualit) # Fit the MCA model to the qualitative data. 

In [11]:
# Using One-Hot Encoding to convert categorical variables in users_qualit into a disjunctive table
one_hot = pd.get_dummies(users_qualit)

mca_no_one_hot = prince.MCA(one_hot=False) # Initializes an MCA model with one_hot=False. Specify that the input data is already one-hot 
                                           # encoded and doesn't require further preprocessing.

mca_no_one_hot = mca_no_one_hot.fit(one_hot) # Fits the MCA model to the one-hot encoded dataset (one_hot). Processes the binary matrix to calculate 
                                             # eigenvalues, coordinates, and other MCA outputs

In [12]:
# Displaying row coordinates
row_coordinates=mca.row_coordinates(users_qualit)
print(row_coordinates)

             0         1
0    -0.223936 -0.391137
1     0.814668 -0.525644
2     0.936202  0.125304
3     0.587992 -0.384968
4    -0.136960  0.225645
...        ...       ...
2995 -0.015425  0.876593
2996 -0.363636  0.366320
2997  0.002740 -0.531813
2998  0.124275  0.119135
2999 -0.223936 -0.391137

[3000 rows x 2 columns]


In [13]:
# Displaying column coordinates
column_coordinates=mca.column_coordinates(users_qualit)
print(column_coordinates)

                         0         1
Country_England   0.363376  0.882161
Country_France   -0.443305 -0.279105
Country_Germany   0.081824 -0.599251
voyage_0         -0.095672 -0.194380
voyage_1          0.335933  0.682523
laugh_0           0.063325 -0.337291
laugh_1          -0.260311  1.386513
photo.elevator_0  0.279769 -0.076954
photo.elevator_1 -1.465151  0.403009
photo.beach_0    -0.322896 -0.002410
photo.beach_1     1.558052  0.011629


In [16]:
# Visualizing the Results
mca.plot(
    users_qualit,
    x_component=0,
    y_component=1,
    show_column_markers=True,
    show_row_markers=False,
    show_column_labels=True,
    show_row_labels=False
)



alt.LayerChart(...)

### MCA Variable Factor Map — Projection of Categories on First Two Dimensions

This graph shows the projection of the **modalities (categories)** of several qualitative variables onto the first two dimensions (axes) obtained from **Multiple Correspondence Analysis (MCA)**.

#### Axes:
- **Component 0 (horizontal axis)** explains **17.88%** of the total variance.
- **Component 1 (vertical axis)** explains **17.26%** of the total variance.
- Together, these two dimensions summarize approximately **35.14%** of the variance in the categorical dataset.

---

#### Interpretation using Coordinates:

Each point represents a modality (e.g. `Country_England`, `voyage_1`, `laugh_0`), positioned using the **column coordinates** extracted earlier.

Some observations:

- **`laugh_1`** (coord: ~[0.063, 1.386]) is located high on **Component 1**, indicating strong contribution to vertical separation. It is far from `laugh_0` (~[-0.260, -0.337]), showing they contrast clearly.
- **`Country_England`** ([0.363, 0.882]) is positioned far from `Country_France` ([-0.443, -0.279]) and `Country_Germany` ([0.081, -0.599]), suggesting these modalities are differentiated along both axes.
- **`photo.beach_0`** and **`photo.beach_1`** are located on opposite sides of **Component 0** (~[-0.323, -0.002] vs. [1.558, 0.011]), indicating a major contrast on the horizontal axis.
- Similarly, **`photo.elevator_1`** ([-1.465, 0.403]) and **`photo.elevator_0`** ([-0.279, -0.076]) also differ significantly, contributing to the spread along **Component 0**.

---

#### General insights:
- Modalities that are **far from the origin** contribute more to the definition of the dimensions and are more discriminating.
- Categories **close together** (e.g. `voyage_0` and `photo.elevator_0`) are likely associated in user profiles.
- Opposing positions (e.g. `photo.beach_0` vs. `photo.beach_1`) suggest that users who select one tend not to select the other.

---

This map helps to interpret the **underlying structure** in the categorical data by revealing how variables and their categories relate and contrast along the principal MCA axes.


In [14]:
## Doing MCA plot with Plotly

# Fit the MCA model
mca = prince.MCA(n_components=2).fit(users_qualit)

# Get the coordinates of the columns (categorical levels)
column_coords = mca.column_coordinates(users_qualit).reset_index()
column_coords.columns = ['label', 'x', 'y']  # Rename for clarity

# Plot with Plotly
fig = px.scatter(
    column_coords,
    x='x',
    y='y',
    text='label',
    title='Figure 2 : MCA of Tinder Users - Qualitative Variables (with Plotly)',
    labels={'x': 'Component 1', 'y': 'Component 2'},
)

# Make labels show up on hover and as text
fig.update_traces(textposition='top center')

# Save the interactive HTML file
fig.write_html("Figure2_MCA_of_tinder_users_qualitative_variables_(with_plotly).html")

# Show the interactive plot
fig.show()

### Figure 2: MCA of Tinder Users — Projection of Qualitative Variables (Plotly)

This graph displays the results of **Multiple Correspondence Analysis (MCA)** applied to a dataset of Tinder users. It shows how the **categories (modalities)** of several qualitative variables are distributed in the reduced two-dimensional MCA space.

---

#### Purpose of the visualization:
The goal is to **explore relationships between categorical variables** by representing them on the first two principal components. These axes capture the most significant variance patterns among the categories.

- **Component 1** (x-axis) and **Component 2** (y-axis) are derived from the MCA and summarize key associations in the data.
- Each dot represents a **category** (e.g. `Country_England`, `photo.beach_1`) based on its contribution to the structure of the dataset.

---

#### Interpretation:

- Categories positioned **far from the center (0,0)** contribute more to the variation captured by these components. For example:
  - `photo.beach_1` and `photo.elevator_1` are located on opposite sides of the plot, suggesting they represent **contrasting user behaviors**.
  - `laugh_1` appears in the upper part of Component 2, indicating it is distinct from `laugh_0`, which is near the origin.

- Categories **close to one another** (e.g. `voyage_0` and `photo.beach_0`) tend to appear together across users, showing possible behavioral patterns.

- The separation between countries (e.g. `Country_England`, `Country_France`, `Country_Germany`) suggests that user preferences or behaviors may differ by country.

---

#### Why this matters:
- This type of plot is used to **identify patterns**, detect **clusters of similar categories**, and support **segment analysis**.
- In a business or research context, such visualizations can support **user profiling**, **market segmentation**, or **survey analysis**.

> The use of Plotly enhances interactivity, making it easier to explore and present the structure of qualitative data in a compact and interpretable way.


[ENG]

3. What do you notice about the inertia represented on this plane?

[FR]

3. Que constatez vous quant a l’inertie représentée sur ce plan ?

In [15]:
# Extract eigenvalues
eigenvalues = mca.eigenvalues_

In [16]:
# Displaying Eigenvalues
print(mca.eigenvalues_summary)


          eigenvalue % of variance % of variance (cumulative)
component                                                    
0              0.215        17.89%                     17.89%
1              0.207        17.26%                     35.15%


In [17]:
# Explained Inertia
eigenvalues = mca.eigenvalues_
total_inertia = sum(eigenvalues)
explained_inertia = [eig / total_inertia for eig in eigenvalues]

print("Explained Inertia:", explained_inertia)

Explained Inertia: [0.5088984457396065, 0.4911015542603935]


In [23]:
# Both dimensions contribute nearly equally to the representation, as the explained inertia for Dimension 1 (50.89%) is only slightly 
# higher than Dimension 2 (49.11%). This balance suggests that the interpretation of the plane requires considering both dimensions together 
# rather than favoring one.

# On the limitations side we can observe the total variance captured by the plane (35.15%) indicates that the first two dimensions only 
# partially summarize the dataset's structure. Consequently 64.85% of the variability is not captured by these two dimensions. 
# Patterns and relationships visible in this plane are meaningful but incomplete, as significant variability remains in the higher dimensions.


### Eigenvalues and Explained Inertia — MCA Interpretation

This step retrieves and analyzes the eigenvalues from the MCA, which represent the amount of variance (or inertia) captured by each principal component.

---

#### Summary of Results:
- **Component 1** explains **17.89%** of the total variance.
- **Component 2** explains **17.26%**.
- The **cumulative variance explained** by the first two components is **35.15%**.

The **explained inertia** (i.e., normalized eigenvalues) shows that:
- Dimension 1 accounts for approximately **50.89%** of the total inertia in the two-component projection.
- Dimension 2 accounts for the remaining **49.11%**.

This near-equal distribution indicates that **both axes contribute significantly** and should be interpreted **together**, rather than favoring a single dominant dimension.

---

#### Interpretation:
- The relatively balanced contribution between the two dimensions enhances the interpretability of the 2D MCA plot, as no axis overwhelmingly dominates the structure.
- However, the first two components together capture only **35.15%** of the total variability in the dataset.
  - This means that **64.85% of the inertia remains in higher dimensions**.
  - Consequently, while the 2D plot provides valuable insights into the structure of the data, it only offers a **partial representation**.

---

#### Implications:
- Patterns and relationships observed in the first two dimensions are meaningful, but **not exhaustive**.
- Any interpretation based solely on this 2D projection should acknowledge that **significant structure may exist in dimensions not shown**.
- For deeper analysis, it may be necessary to examine **additional components**, especially if cumulative inertia remains low.

> In summary, the current factorial plane is informative but limited. Both axes contribute almost equally and should be considered jointly, but additional dimensions may be required for a more comprehensive view of the data structure.


[ENG]

4. Notwithstanding the value of the variance associated with the first two axes, try to write down without a numerical value what you see in the last figure, in terms of a macrosocial phenomenon. The aim here is to bring out the typical profiles of dating application users.

[FR]

4. Nonobstant la valeur de la variance associée aux deux premiers axes, tâchez de d´ecrire sans valeur chiffrée ce que vous visualisez dans la dernière figure, en termes de phénomène macrosocial. Il s’agit ici de faire émerger des formes de profils-type d’utilisateurs d’application de dating.

### Interpretation of MCA Plot — Sociological Profiles of Dating App Users

Although the first two dimensions only capture a limited portion of the total variance, the spatial distribution of categories still allows for a preliminary identification of **typical user profiles** based on their preferences and behaviors. These groupings may reflect broader sociocultural patterns in how users present themselves on dating platforms.

---

#### 1. **Profile 1 — Reserved or Serious-Minded Users (France)**
The categories `voyage_0`, `laugh_0`, and `Country_France` appear clustered in the same region of the MCA space. This suggests a user profile associated with:
- Living in France,
- A lower tendency to highlight humor or travel in their profile.

This may reflect a group of users who approach dating more seriously or conservatively, possibly emphasizing stability or depth over casual or playful presentation.

---

#### 2. **Profile 2 — Social and Adventurous Users (England)**
The proximity of `voyage_1` and `Country_England` suggests another user segment characterized by:
- A strong interest in travel,
- A likely inclination to share travel-related content in their profiles,
- Residence in the United Kingdom.

These individuals may associate dating with exploration and shared experiences, and value openness or dynamism in their interactions.

---

#### 3. **Profile 3 — Artistic or Unconventional Aesthetic**
The variable `photo.elevator_1` is positioned distinctly from the others, indicating a smaller cluster of users who:
- Choose to include photos taken in elevators, possibly as a form of creative or unconventional self-presentation.

This group may reflect users who aim to differentiate themselves through aesthetic or artistic choices, possibly appealing to niche interests or non-conformist identities.

---

#### 4. **Profile 4 — Appearance-Focused or Leisure-Oriented**
The position of `photo.beach_1` identifies a category of users who:
- Highlight beach photographs in their profile, suggesting an emphasis on leisure, physical appearance, or lifestyle branding.

This may point to a user group more concerned with visual appeal, self-image, or health-oriented interests, often associated with a more extroverted or socially active profile.

---

### Conclusion
These spatial proximities between categorical variables provide insights into emerging **macrosocial user segments** on the dating platform. While interpretations should remain cautious due to the partial variance explained by the first two dimensions, the MCA results already suggest meaningful clusters that align with lifestyle, cultural background, and self-presentation strategies.


[ENG]

5. Represent the contribution of the variables on the first two axes using a table.

[FR]

5. Représentez la contribution des variables sur les deux premiers axes via une table.

In [18]:
# Calculate contributions
contributions = pd.DataFrame()
for i in range(column_coordinates.shape[1]):  # Loop over dimensions
    dimension = column_coordinates.iloc[:, i]
    contributions[f'Contribution to Dimension {i+1}'] = (dimension ** 2) / eigenvalues[i]

print(contributions)

                  Contribution to Dimension 1  Contribution to Dimension 2
Country_England                      0.615082                     3.756438
Country_France                       0.915431                     0.376026
Country_Germany                      0.031187                     1.733400
voyage_0                             0.042638                     0.182383
voyage_1                             0.525684                     2.248613
laugh_0                              0.018680                     0.549149
laugh_1                              0.315649                     9.279590
photo.elevator_0                     0.364602                     0.028585
photo.elevator_1                     9.999665                     0.783989
photo.beach_0                        0.485676                     0.000028
photo.beach_1                       11.307975                     0.000653


### Contribution of Variables to the First Two Dimensions — MCA Interpretation

This step computes the contribution of each categorical variable (modality) to the first two principal dimensions obtained from the Multiple Correspondence Analysis (MCA). These contributions help quantify **how much each category influences** the structure of each axis.

---

#### Method:
- Contributions are calculated using the formula:
  $$
  \text{Contribution}_{i,j} = \frac{{\text{coordinate}_{i,j}^2}}{{\text{eigenvalue}_j}}
  $$
  where `i` is the variable and `j` is the dimension.
- This formula captures how strongly a given category influences the geometric structure of a particular component.
- The higher the contribution, the more the category helps define that axis.

---

#### Key Observations:

**Dimension 1 (horizontal axis):**
- The categories with the highest contributions are:
  - `photo.beach_1` → **11.31** (very strong influence),
  - `photo.elevator_1` → **1.00**,
  - `Country_France` → **0.92**.
- These categories are primary drivers of variation along the first component and are central to interpreting it.

**Dimension 2 (vertical axis):**
- The most influential categories are:
  - `laugh_1` → **9.28**,
  - `Country_England` → **3.76**,
  - `voyage_1` → **2.25**.
- This suggests that **humor, travel preference, and nationality** are important factors shaping the second dimension.

---

#### Why this matters:
- Identifying which categories contribute the most to each dimension allows for a **deeper interpretation** of the MCA axes.
- This step reveals that:
  - **Dimension 1** is likely defined by **visual presentation choices** (e.g., photos at the beach or in elevators).
  - **Dimension 2** captures **personal values and lifestyle**, such as interest in humor and travel.

> These findings help link the mathematical output of MCA to **real-world user behaviors and profiles**, improving the interpretability of the factor maps and supporting user segmentation.
